src/node_sqlite.cc

```diff
@@ -1763,21 +1763,25 @@ void DatabaseSync::ApplyChangeset(const FunctionCallbackInfo<Value>& args) {
 
       Local<Function> filterFunc = filterValue.As<Function>();
 
-      context.filterCallback = [env,
-                                filterFunc](std::string_view item) -> bool {
-        // TODO(@jasnell): The use of ToLocalChecked here means that if
-        // the filter function throws an error the process will crash.
-        // The filterCallback should be updated to avoid the check and
-        // propagate the error correctly.
-        Local<Value> argv[] = {
-            String::NewFromUtf8(env->isolate(),
-                                item.data(),
-                                NewStringType::kNormal,
-                                static_cast<int>(item.size()))
-                .ToLocalChecked()};
-        Local<Value> result =
-            filterFunc->Call(env->context(), Null(env->isolate()), 1, argv)
-                .ToLocalChecked();
+      context.filterCallback =
+          [db, env, filterFunc](std::string_view item) -> bool {
```

Comment:
```suggestion
          [&](std::string_view item) -> bool {
```

?

Since this is synchronously called, it should be fine to just use a refrence.

Passing these capures by value makes it look like the function is asynchronously called, but if that were the case, this would actually be a bug, because then there's a chance that `db` might get garbage collected before the function gets called (our solution for that would be to use `BaseObjectPtr`s, if you're curious).

Reply:
Oh, that makes so much sense.

Reply:
Nice catch.